### ISSUE TITLE
FIX: Non standard dialect specific messages

### ISSUE THREAD

**Issue by divaykin** [libtrading/libtrading#99]
What's the best way to send non standard dialect specific message?

The problem is https://github.com/libtrading/libtrading/blob/master/include/libtrading/proto/fix_message.h#L16

There is only one generic enumeration with all the message types.

**Comment by penberg** [comment]
@divaykin I think we need to change the APIs to allow passing an arbitrary string as message type and have a lookup function for translating from the enum to a message type string. @mstanichenko, comments?

**Comment by mstanichenko** [comment]
I believe that we may change `struct fix_message` in the following way. It is like this right now:
```C
struct fix_message {
    enum fix_msg_type type;
...
}
```

I suggest that we should change it to
```C
struct fix_message {
    const char *type;
...
}
```

If it would be like this we had not any problems with sending, hadn't we?
AFAICT, there are no problems with receiving messages as well. We just let users to call `fix_msg_type_parse` on their own or any other dialect-specific function which will handle it.

**Comment by penberg** [comment]
@mstanichenko Makes sense. Wanna cook up a patch? :wink:

**Comment by mstanichenko** [comment]
Will do

**Comment by penberg** [comment]
Fixed by #102. Closing.
